fred2/briefingeditordlg.cpp

```diff
@@ -1090,31 +1093,38 @@ void briefing_editor_dlg::OnMakeIcon()
         else
             biconp->type = ICON_ASTEROID_FIELD;
 	}
-	// jumpnodes
-	else if(jnp != Jump_nodes.end()){
-		// find the first navbuoy
+	// jumpnodes -- have to use Navbuoy if clicked on during briefing because jump nodes are not in ship_info
+	else if (jump_node_objnum > -1) {
+		// find the first navbuoy, by iterating through the ship classes
 		biconp->ship_class = -1;
-        for (auto it = Ship_info.cbegin(); it != Ship_info.cend(); ++it)
-        {
-            if (it->flags[Ship::Info_Flags::Navbuoy])
+		
+		int i = 0;
+		for (ship_info &it : Ship_info)
+		{
+            if (it.flags[Ship::Info_Flags::Navbuoy])
             {
-                iconp->ship_class = (int)std::distance(Ship_info.cbegin(), it);
+                biconp->ship_class = i;
                 break;
             }
+			i++;
```

Comment:
You shouldn't increment an index every iteration of a for loop unless you actually need to *use* it in every iteration. `std::distance()` is a perfectly fine function.